src/index.js

```diff
@@ -1,37 +1,114 @@
 import './style.css';
 
-const taskContainer = document.getElementById('tasks');
-
-const toDoArr = [
-  {
-    desc: 'I will clean my room',
-    bool: 'true',
-    index: 0,
-  },
-  {
-    desc: 'I have to read a book',
-    bool: 'true',
-    index: 1,
-  },
-  {
-    desc: 'Play fifa with Camavinga',
-    bool: 'false',
-    index: 2,
-  },
-];
-
-const getTasks = () => {
-  toDoArr.forEach((activity, index) => {
-    taskContainer.innerHTML += `
-      <li class="activity">
-        <div class="checkBoxex">
-          <input type="checkbox" name="action1" id="${index}"/>
-          <label for="${index}">${activity.desc}</label><br>
+const form = document.getElementById('todoform');
+const todoInput = document.getElementById('newtodo');
+const toDoListEl = document.getElementById('todos-list');
+let todos = JSON.parse(localStorage.getItem('todos')) || [];
+let EditTodoId = -1;
+
+const renderTodos = () => {
+  toDoListEl.innerHTML = '';
+
+  todos.forEach((todo, index) => {
+    toDoListEl.innerHTML += `
+        <div class="todo" id="${index}">
+            <i 
+              class="bi ${todo.checked ? 'bi-check-circle-fill' : 'bi-circle'}"
+              style="color : ${todo.color}"
+              data-action="check"
+              ></i>
+            <p class="" data-action="check">${todo.value}</p>
+            <i class="bi bi-pencil-square" data-action="edit"></i>
+            <i class="bi bi-trash" data-action="delete"></i>
         </div>
-        <i class="fa-solid fa-ellipsis-vertical fs-3 "></i>
-      </li>
-    `;
+        `;
   });
 };
 
-getTasks();
\ No newline at end of file
+renderTodos();
+
+// eslint-disable-next-line consistent-return
+const saveTodo = () => {
+  const todoValue = todoInput.value;
+
+  const isEmpty = todoValue === '';
+
+  const isDuplicate = todos.some((todo) => todo.value.toUpperCase() === todoValue.toUpperCase());
+
+  if (isEmpty) {
+    return '';
+  } if (isDuplicate) {
+    return '';
+  }
+  if (EditTodoId >= 0) {
+    todos = todos.map((todo, index) => ({
+      ...todo,
+      value: index === EditTodoId ? todoValue : todo.value,
+    }));
+    EditTodoId = -1;
+  } else {
+    todos.push({
+      value: todoValue,
+      checked: false,
+      color: 'black',
+    });
+  }
+
+  todoInput.value = '';
+};
+
+form.addEventListener('submit', (event) => {
+  event.preventDefault();
+
+  saveTodo();
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+});
+
+function checkTodo(todoId) {
+  todos = todos.map((todo, index) => ({
+    ...todo,
+    checked: index === todoId ? !todo.checked : todo.checked,
+  }));
+
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
+
+function editTodo(todoId) {
+  todoInput.value = todos[todoId].value;
+  EditTodoId = todoId;
+}
+
+function deleteTodo(todoId) {
+  todos = todos.filter((todo, index) => index !== todoId);
+  EditTodoId = -1;
+  renderTodos();
+  localStorage.setItem('todos', JSON.stringify(todos));
+}
```

Comment:
- [ ] As per the project requirements, it would be great if you could make sure that when a task is deleted, the corresponding index is also updated. Currently, the app is doing a great job deleting tasks, but the index doesn't seem to update along with it. Thank you for looking into this! 😊👍
   
   [scrnli_18_8_2023_1-06-23.webm](https://github.com/Chimoneg27/To-Do-List/assets/107952724/cd1e923a-f9d3-42fc-a9d0-093bc0998c2d)

   ![image](https://github.com/Chimoneg27/To-Do-List/assets/107952724/63de1f7b-fe8d-4fa8-846b-41960fe1df36)